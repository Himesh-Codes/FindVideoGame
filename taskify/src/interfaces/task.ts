import { FormEvent } from "react";

export interface TaskState {
    task: string;
    setTask: React.Dispatch<React.SetStateAction<string>>;
    addTasks: (event: FormEvent<Element>) => void;
}

export interface Task {
    id: string;
    taskName: string;
    isDone?: boolean;
}

export interface TaskListProp{
    tasks?: Task[];
    pushStacks: React.Dispatch<React.SetStateAction<Task[]>>;
}