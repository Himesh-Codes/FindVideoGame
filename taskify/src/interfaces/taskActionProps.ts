export interface DeleteTasksProp {
    taskId: string;
}

export interface CompleteTasksProp {
    taskId: string;
}

export interface EditTasksProp {
    taskId: string;
    editContent: string;
}

export interface AddTasksProp{
    recentTaskUid: string;
    taskName: string;
}

export interface TaskActions {
    type: 'add' | 'delete' | 'complete' | 'edit';
    action: DeleteTasksProp | AddTasksProp;
}