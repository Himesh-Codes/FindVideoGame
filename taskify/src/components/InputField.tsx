import {useRef } from "react";
import { TaskState } from "../interfaces/task";
import "../App.css";

export default function InputField(inputProps: TaskState): React.JSX.Element{
    const inputFieldActive = useRef<HTMLInputElement>(null);

    return(
        <form className="formContainer" onSubmit={(event)=> HandleSubmit(inputProps, event, inputFieldActive)}>
            <div className="addTask">
                <input ref={inputFieldActive} type="text" placeholder="Enter the task" value={inputProps.task} onChange={(event)=> inputProps.setTask(event.target.value)} className="inputField"/>
                <button className="addTaskBtn" type="submit">ADD</button>
            </div>
        </form>
    );
}

function HandleSubmit(inputProps: TaskState, event: React.FormEvent, inputFieldActive: React.RefObject<HTMLInputElement>){
    inputProps.addTasks(event);
    inputFieldActive.current?.blur();
}