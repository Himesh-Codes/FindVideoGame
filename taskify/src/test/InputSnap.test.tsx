import * as renderer from 'react-test-renderer';
import InputField from '../components/InputField';
import { TaskState } from '../interfaces/task';

// snapshot validation on field UI changes
it('input field UI changes validation', () => {
    const mockCallback = jest.fn();
    const inputProps: TaskState = {task: "", setTask: ()=>{}, addTasks: mockCallback};
    const inputComponentTree = renderer.create(
        <InputField {...inputProps} />
    ); //render the component tree created
    let treeJsom = inputComponentTree.toJSON(); // convert tree to JSON
    expect(treeJsom).toMatchSnapshot(); //expect the snapshot comparison

    //functional test with submit
   renderer.act(() => {
      (treeJsom as renderer.ReactTestRendererJSON).props.onSubmit(); // Simulate the form submit
    });

    // rerender the component
    treeJsom = inputComponentTree.toJSON();
    // Assert that the mock event object was called with the expected arguments
    expect(mockCallback).toHaveBeenCalledTimes(1);
});