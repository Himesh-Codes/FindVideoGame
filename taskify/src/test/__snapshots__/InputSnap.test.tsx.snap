// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input field UI changes validation 1`] = `
<form
  className="formContainer"
  onSubmit={[Function]}
>
  <div
    className="addTask"
  >
    <input
      className="inputField"
      onChange={[Function]}
      placeholder="Enter the task"
      type="text"
      value=""
    />
    <button
      className="addTaskBtn"
      type="submit"
    >
      ADD
    </button>
  </div>
</form>
`;
